bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -1205,6 +1225,9 @@ void RewriteInstance::discoverFileObjects() {
     PreviousFunction = BF;
   }
 
+  if (BC->IsLinuxKernel && !std::get<0>(BC->LinuxKernelVersion))
+    BC->errs() << "BOLT-WARNING: Linux kernel version is unknown\n";
```

Comment:
If the version of the kernel is unknown, you just emit a warning instead of erroring out?

Reply:
If BOLT errors out here, then all Linux kernel related tests need to specify a proper version.

It would be better to do that latter when functions like `readAltInstructions()` and `readStaticKeysJumpTable()` have been modified to utilize `LinuxKernelVersion` to handle differences between Linux kernel versions.

Reply:
I've added a Linux kernel version for each related test and changed the behavior to erroring out.